pkg/odo/cli/component/push.go

```diff
@@ -140,6 +140,10 @@ func (po *PushOptions) Complete(name string, cmd *cobra.Command, args []string)
 		// The namespace was retrieved from the --project flag (or from the kube client if not set) and stored in kclient when initalizing the context
 		if !pushtarget.IsPushTargetDocker() {
 			po.namespace = po.KClient.Namespace
+			// log warning in case of default namespace
+			if po.namespace == "default" {
+				log.Warning("odo may not work as expected in default namespace, please run odo component in non default namespace")
```

Comment:
In v1, we used to create a `myproject` namespace for the user if they were working on a fresh cluster (which is when I think we will most likely face the issue we're trying to address here.)

If we don't want to do that, we should add something in the Warning that says that they can use `odo project create` to create a new namespace/project. Also, aren't we using the word "project" for odo users instead of "namespace"?

Reply:
> Warning that says that they can use odo project create to create a new namespace/project

+1 for mentioning that user should create a new project using `odo project create`



> Also, aren't we using the word "project" for odo users instead of "namespace"?

-1 we should be using "project" everywhere (https://github.com/openshift/odo/issues/3947#issuecomment-692789890)

Reply:
@dharmit @kadel Please check, is it look fine now?